grpc/pb-marshalling.go

```diff
@@ -115,7 +116,8 @@ func pbToChallenge(in *corepb.Challenge) (challenge core.Challenge, err error) {
 	if in == nil {
 		return core.Challenge{}, ErrMissingParameters
 	}
-	if in.Id == nil || in.Type == nil || in.Status == nil || in.Token == nil || in.KeyAuthorization == nil {
+	newAuthzSchema := features.Enabled(features.NewAuthorizationSchema)
```

Comment:
I think pb-marshalling is so low-level I'd prefer not to make it depend on feature flags. In this case, since this is just an extra layer of checking, I think we don't need to gate on `NewAuthorizationSchema`. Instead we can just start unconditionally saying nil challenge `Id` is okay.

Reply:
Done, note this also means allowing nil `KeyAuthorization`.